docs/dev/adding_instruments.rst

```diff
@@ -442,6 +447,38 @@ The dictionary now maps the keys to specific values. The values and keys can be
 
 As you have seen, the :func:`Instrument.control <pymeasure.instruments.Instrument.control>` function can be significantly extended by using validators and maps.
 
+Boolean properties
+******************
+
+The idea of using maps can be leveraged to implement properties where the user-facing values are booleans, so you can interact in a pythonic way using :code:`True` and :code:`False`:
+
+.. testcode::
+
+    Extreme5000.output_enabled = Instrument.control(
+        "OUTP?", "OUTP %d",
+        """Control the instrument output is enabled (boolean).""",
+        map_values=True,
+        values={True: 1, False: 0},  # the dict values could also be "on" and "off", etc.
+    )
+
+
+.. doctest::
+
+    >>> extreme = Extreme5000("GPIB::1")
+    >>> extreme.output_enabled = True
+    >>> extreme.read()
+    '1'
+    >>> extreme.output_enabled = False
+    >>> extreme.output_enabled
+    False
+    >>> # Invalid input raises an exception
+    >>> extreme.output_enabled = 34
+    Traceback (most recent call last):
+    ...
+    KeyError: 34
```

Comment:
So, I realised that we don't need the `strict_discrete_set` validator for the boolean property -- if an invalid input is supplied, you get a KeyError from the dict lookup as shown in the example.

Do you think we should leave it in the docs like this, or with the `strict_discrete_set` added, as the error message might be clearer/more useful?

Reply:
I personally prefer cleaner code (with less additions).

However, a validator ensures the right values. 

We could add, that a mapping dictionary works as a validator as well (maybe even transform the exception in `values` from KeyError to Value Error("Invalid value supplied for mapping") or something.

This should enter the docstring and documentation, I'd say.

Reply:
> We could add, that a mapping dictionary works as a validator as well (maybe even transform the exception in values from KeyError to Value Error("Invalid value supplied for mapping") or something.

That's probably more confusing to users. Let's add the `strict_discrete_set` back, the error message is clearer, and semantically it's also clearer that the inputs are being validated (even if they would also be "validated" in a way by the subsequent dict lookup).

Reply:
I put the validator in now. Agreed to do it that way?
(Note: I know that the error message talks about a set but displays a dict, but that's a battle for another day)

Reply:
In the end, it is an example how you can do it. With the validator in the example, the readers will see how to validate correctly.

So, I'm ok with the validator.